src/gsi-ncdiag/gsi_ncdiag.py

```diff
@@ -1291,310 +1326,89 @@ def toIODAobs(self, OutDir, ObsBias, QCVars, TestRefs, clobber=True):
+                gvname = "brightness_temperature", iodavar
+                outdata[gvname] = np.reshape(tmp, (nlocs, nchans))
+                VarDims[gvname] = ['nlocs', 'nchans']
+
+        # brightness temperature variables
+        value = 'brightness_temperature'
+        obsdata[obsdata > 9e5] = self.FLOAT_FILL
+        obsqc[obsdata > 9e5] = self.INT_FILL
+
+        # store values in output data dictionary
+        outdata[varDict[value]['valKey']] = np.reshape(obsdata, (nlocs, nchans))
+        outdata[varDict[value]['errKey']] = np.reshape(obserr, (nlocs, nchans))
+        outdata[varDict[value]['qcKey']] = np.reshape(obsqc.astype(np.int32), (nlocs, nchans))
+        # create a GSI effective QC variable
+        gsiqcname = value, 'GsiEffectiveQC'
+        errname = value, 'GsiFinalObsError'
+        gsiqc = np.zeros_like(outdata[varDict[value]['valKey']])
+        gsiqc[outdata[errname] > 1e8] = 1
+        gsiqc[np.reshape(self.var('QC_Flag'), (nlocs, nchans)) < 0] = 1
+        outdata[gsiqcname] = gsiqc.astype(np.int32)
+        varAttrs[gsiqcname]['units'] = 'unitless'
 
-                    # store values in output data dictionary
-                    outdata[varDict[value]['bctKey']] = obsbiastermsub
-                    outdata[varDict[value]['bcpKey']] = obsbiaspredsub
-                    ii += 1
+        if (ObsBias):
+            valuebc = [
+                "constant",
+                "zenith_angle",
+                "cloud_liquid_water",
+                "lapse_rate_squared",
+                "lapse_rate",
+                "cosine_of_latitude_times_orbit_node",
+                "sine_of_latitude",
+                "emissivity",
+                "scan_angle_order_4",
+                "scan_angle_order_3",
+                "scan_angle_order_2",
+                "scan_angle",
+            ]
+            ii = 0
+            for value in valuebc:
+                obsbiastermsub = obsbiasterm[ii]
+                obsbiaspredsub = obsbiaspred[ii]
+                obsbiastermsub[obsbiastermsub > 9e5] = self.FLOAT_FILL
+                obsbiaspredsub[obsbiaspredsub > 9e5] = self.FLOAT_FILL
+
+                # store values in output data dictionary
+                outdata[varDict[value]['bctKey']] = obsbiastermsub
+                outdata[varDict[value]['bcpKey']] = obsbiaspredsub
+                if valuebc in units_values.keys():
+                    varAttrs[varDict[value]['bctKey']]['units'] = units_values[valuebc]
+                    varAttrs[varDict[value]['bptKey']]['units'] = units_values[valuebc]
+                ii += 1
         # var metadata
-        var_mdata['variable_names'] = writer.FillNcVector(var_names, "string")
         for key, value2 in chan_metadata_dict.items():
             try:
-                var_mdata[value2] = self.var(key)
+                if value2 in chan_metadata_int:
+                    outdata[(value2, 'VarMetaData')] = self.var(key).astype(np.int32)
```

Comment:
The group for these variables should be `MetaData` since we've deprecated the `VarMetaData` group in v2. The dimensions can remain the same and it's the dimension that identifies what that particular variable describes. Eg, dimension is nlocs, the variable describes locations; dimension is nchans, the variable describes channels.

Same for the occurrences of `VarMetaData` below.

Thanks!